File: backend/ambulance-service/src/main/java/com/example/ambulanceservice/model/Ambulance.java
package com.example.ambulanceservice.model;

import com.example.ambulanceservice.model.enums.AmbulanceStatus;
import com.example.ambulanceservice.model.enums.Specialty;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "ambulances")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Ambulance {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String registrationNumber;

    private String model;

    @Enumerated(EnumType.STRING)
    private AmbulanceStatus status;

    private double latitude;

    private double longitude;

    private Specialty specialty;

    private String phone;

}

File: mobile/app/src/main/java/com/example/ambulanceapp/beans/Ambulance.kt
package com.example.ambulanceapp.beans

import org.simpleframework.xml.Element
import org.simpleframework.xml.Root

@Root(name = "ambulance")
data class Ambulance(
    @field:Element(name = "id")
    var id: Long = 0,

    @field:Element(name = "registrationNumber")
    var registrationNumber: String? = null,

    @field:Element(name = "model")
    var model: String? = null,

    @field:Element(name = "status")
    var status: String? = null,

    @field:Element(name = "latitude")
    var latitude: Double = 0.0,

    @field:Element(name = "longitude")
    var longitude: Double = 0.0,

    @field:Element(name = "specialty")
    var specialty: String? = null,

    @field:Element(name = "phone")
    var phone: String? = null,
)